src/core/options-watcher.h

```diff
@@ -0,0 +1,55 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2023 Intel Corporation. All Rights Reserved.
+#pragma once
+
+
+#include <librealsense2/h/rs_option.h>
+#include <src/core/option-interface.h>
+
+#include <rsutils/signal.h>
+#include <rsutils/concurrency/concurrency.h>
+
+#include <map>
+#include <chrono>
+#include <functional>
+#include <memory>
+#include <thread>
+
+
+namespace librealsense {
+
+
+// Automatically updates (queries) registered options values according to set time intervals
```

Comment:
I don't like this.
So the purpose is **_NOT_** to notify? I.e., even without a subscription, the point is a periodic query? Why?

Reply:
I will rephrase. We don't query if no one is registered.

Reply:
Done